src/lib/apiTypesV2/Subscription.h

```diff
@@ -0,0 +1,78 @@
+#ifndef SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+#define SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include <string>
+#include <vector>
+
+#include "../ngsi/Duration.h"
+#include "../ngsi/Throttling.h"
+
+struct EntID
+{
+  std::string id;
+  std::string idPattern;
+  std::string type;
+  std::string toJson();
+};
```

Comment:
I agree to your comment about carrying a std::vector that may be empty.
We started to use std::string and std::vector because they make the code so much easier to read and even implement. My first choice was to use a pointer, of course, and allocate the vector when needed but that made the code harder to read (much harder), so vectors are used.
Son't expect to see the code 100% perfect, all over. It is not perfect. But let's not use that fact as an excuse to not trying to make the new code as good as possible.

Perhaps we can try to make a new Entity class that it good enough for all three uses, or we maintain three different classed but with some prefix to clearly show what is the use of each one.

Right now we have   ... EntityId, Entity and EntID ... just too confusing.